packages/babel-plugin-transform-parameters/test/fixtures/parameters/iife-this-9385/output.js

```diff
@@ -0,0 +1,35 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.Test = void 0;
+
+function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
+
+function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
+
+function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }
+
+var Test =
+/*#__PURE__*/
+function () {
+  function Test() {
+    _classCallCheck(this, Test);
+  }
+
+  _createClass(Test, [{
+    key: "invite",
+    value: function invite() {
+      var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
+      return function () {
```

Comment:
Why is the wrapper generated?

Reply:
Babel deopts because it thinks that the `privacy` key in the object type is a reference. I opened a PR to fix the iife case, but I'll see how we can avoid the deoptimization here.

Reply:
Will be fixed by #9605